misoc/cores/icap.py

```diff
@@ -0,0 +1,126 @@
+from migen import *
+from migen.genlib.cdc import PulseSynchronizer
+
+from misoc.interconnect.csr import AutoCSR, CSR
+
+class ICAP(Module, AutoCSR):
+    def __init__(self, fpga_family, clk_divide_ratio="2"):
+        """
+        ICAP module.
+
+        Use this module to issue the IPROG command and restart the gateware.
+        Both E2 and E3 are supported by selecting the right version.+
+        
+        Parameters
+        ----------
+        flga_family : str
+            FPGA family name, used to determine the version of primitive. 
+            Supported family: xcku040 (metlino/sayma), xc7a100t (kasli)
+
+        clk_divide_ratio : str
+            Optional. The divide ratio of the clock frequency from system clock.
+        """
+        self.iprog = CSR()
+
+        ###
+        if fpga_family not in {"xcku040", "xc7a100t"}:
+            raise ValueError("Not supported FPGA family")
+
+        iprog_command_seq_i = [
```

Comment:
Why _i?

Reply:
Because it is the inverted version of the original command

Reply:
Ok.